### PR TITLE
Add changes to Menu to the CHANGELOG

### PR DESCRIPTION
**Pull request by LennertBlommaert**
### Description

The main goal of the [refactor of the `Menu` component](https://github.com/teamleadercrm/ui/pull/330), was to remove its soon to be deprecated lifecycles.

I forgot to add these changes to the `CHANGELOG`, hence this extra PR.

#### Screenshot before this PR

Not applicable.

#### Screenshot after this PR

Not applicable.

### Breaking changes

None.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 - `DataGrid`: the `componentWillReceiveProps` lifecycle is replaced, as it will be decrecated in react v17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#331](https://github.com/teamleadercrm/ui/pull/331))
 - `Overlay`: the `componentWillUpdate` lifecycle is replaced, as it will be decrecated in react v17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#331](https://github.com/teamleadercrm/ui/pull/331))
+- `Menu`: the `componentWillUpdate` and `componentWillRecieveProps` lifecycles are replaced, as they will be decrecated in react v17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#330](https://github.com/teamleadercrm/ui/pull/330))
```
Comment (duivvv) [inline CHANGELOG.md]: A couple of remarks:

- there is `decrecated` 3 times in this file (typo)
- shouldn't we provide a bit more info? Replaced in favour of what?
- `React 17` not `react v17`

Comment (LennertBlommaert) [reply CHANGELOG.md]: Something like:
`Menu`: the `componentWillUpdate` and `componentWillRecieveProps` lifecycles are replaced by the `getDerivedStateFromProps` and `componentDidUpdate` lifecycles, as they will be deprecated in React 17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#330](https://github.com/teamleadercrm/ui/pull/330))

For providing a bit more info?

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,9 +10,9 @@
...
-- `Overlay`: the `componentWillUpdate` lifecycle is replaced, as it will be deprecated in react v17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#331](https://github.com/teamleadercrm/ui/pull/331))
-- `Menu`: the `componentWillUpdate` and `componentWillRecieveProps` lifecycles are replaced, as they will be deprecated in react v17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#330](https://github.com/teamleadercrm/ui/pull/330))
+- `DataGrid`: the `componentWillReceiveProps` lifecycle is replaced, as it will be deprecated in React 17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#331](https://github.com/teamleadercrm/ui/pull/331))
+- `Overlay`: the `componentWillUpdate` lifecycle is replaced, as it will be deprecated in React 17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#331](https://github.com/teamleadercrm/ui/pull/331))
+- `Menu`: the `componentWillUpdate` and `componentWillRecieveProps` lifecycles are replaced, as they will be deprecated in React 17 ([@LennertBlommaert](https://github.com/LennertBlommaert) in [#330](https://github.com/teamleadercrm/ui/pull/330))
```
Comment (nickwaelkens) [inline CHANGELOG.md]: `componentWillRecieveProps` should be `componentWillReceiveProps`

File: CHANGELOG.md (hunk 1 above)
Comment (nickwaelkens) [reply CHANGELOG.md]: What @duivvv means is also add what it has changed into; `componentWillReceiveProps` ➡️ `aBetterLifecycleMethod`

Comment (duivvv) [reply CHANGELOG.md]: Yeah either one is fine, just a bit more info. @nickwaelkens solution might be a bit more concise.

**Comment by lowiebenoot** [comment]
Just thinking: shouldn't this be under the "changed" section?

**Comment by LennertBlommaert** [comment]
Maybe it should only be a fix, if the lifecycles were already deprecated. But on the other hand is it a change if nothing changed for the end user of the package?

**Comment by lowiebenoot** [comment]
@LennertBlommaert yes, I thought about that as well after I posted the comment 😅 

FYI: there's a conflict.

**Pull request merged**